sdk/storage/azure-storage-common/inc/azure/storage/common/paged_response.hpp

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <cstdlib>
+#include <string>
+
+#include <azure/core/context.hpp>
+#include <azure/core/http/raw_response.hpp>
+
+namespace Azure {
+
+template <class Derived> class PagedResponse {
+public:
+  std::string CurrentPageToken;
+  std::string NextPageToken;
+  std::unique_ptr<Azure::Core::Http::RawResponse> RawResponse;
+
+  bool HasMorePages() const { return m_hasMorePages; }
+
+  void MoveToNextPage(const Azure::Core::Context& context = Azure::Core::Context())
+  {
+    static_assert(
+        std::is_base_of<PagedResponse, Derived>::value,
+        "The template argument \"Derived\" should derive from PagedResponse<Derived>.");
+
+    if (NextPageToken.empty())
```

Comment:
```suggestion
    if (!m_hasMorePages || NextPageToken.empty())
```

Reply:
The check doesn't crash or throw.

Reply:
I'll just leave it undefined behavior. Like when you access out of bound of a vector or increment an iterator even it's equal to end().

Maybe core sdk guys can fix it when moving it into core sdk.

Reply:
https://github.com/Azure/azure-sdk-for-cpp/issues/2118#issuecomment-820954804